torch/csrc/jit/codegen/cuda/python_frontend/README.md

```diff
@@ -0,0 +1,119 @@
+# nvFuser Python Frontend
+
+This frontend allows for a user to describe the set of operations for nvFuser to fuse via 1 or more kernels.  This frontend is intended to be an integration point with PyTorch or standalone applications.
+
+# Usage
+
+## Example
+
+```python
+import torch
+from torch._C._nvfuser import FusionManager, FusionDefinition, DataType
+
+fm = FusionManager.get()
+with FusionDefinition(fm) as fd :
+    t0 = fd.define_tensor(symbolic_sizes=[-1, 1, -1],
+                          contiguous=[True, True, True],
+                          dtype=DataType.Float)
+    t1 = fd.define_tensor(3)
+    c0 = fd.define_constant(3.0)
+
+    t2 = fd.ops.add(t0, t1)
+    t3 = fd.ops.mul(t2, c0)
+    t4 = fd.ops.sum(t3, [-1], False, DataType.Float)
+
+    fd.add_output(t4)
+
+input1 = torch.ones(2, 1, 8, device='cuda')
+input2 = torch.ones(2, 4, 8, device='cuda')
+
+nvf_out1 = fm.execute([input1, input2])[0]
+```
+
+## Components
+
+### `FusionManager` - Executes and Caches Defined Fusions
+#### `FusionManager` Methods
+* `get()`: Provides a global instance.  You cannot directly construct an instance of the `FusionManager`.
+* `execute([inputs])`:  Allows you to execute the currently defined fusion with a list of given inputs.
+* `print_ir()`: Prints the low level IR for the currently defined fusion.
+
+### `FusionDefiniton` Context Manager - Interface for Defining Fusions
+
+#### Defining Input Tensors
+_All intermediate tensors are created by operations.  Constant tensors do not exist._
```

Comment:
Will the support of tensor factories cover many needs for "constant tensors"? I will work on `full(value, shape)`, as well as `zeros` and `ones`, after arange https://github.com/csarofeen/pytorch/pull/1933

Reply:
We might have to revisit `define_constant` to support a vector of constants.